### ISSUE TITLE
Create Retrieval Reranker Model

### ISSUE THREAD

**Issue by marctjones** [marctjones/klareco#30]
**Part of**: M1 Foundation (before training data generation)
**Duration**: 2-3 weeks
**Priority**: P1
**Depends on**: Enhanced corpus with embeddings (#28)

## Objective

Train a lightweight reranker model to score (query, candidate) relevance for retrieval.

## Background

Currently using semantic similarity from Tree-LSTM encoder. A dedicated reranker will improve retrieval quality for Q&A.

## Architecture

See [[Deep-Learning-Models-Reference#Model-3-Retrieval-Reranker]] for full specs.

```python
class RetrievalReranker(nn.Module):
    encoder: TreeLSTMEncoder  # 1.8M params (shared)
    interaction_mlp: MLP      # ~100K params
    # Total: ~1.9M parameters
```

## Training Data

Generate from corpus:
- **Synthetic Q&A pairs**: Use templates to create (question, answer) pairs
- **Positive examples**: (question, correct_answer) → score 1.0
- **Negative examples**: (question, random_sentence) → score 0.0

**Data volume**: 50K-100K pairs

## Tasks

- [ ] Create `klareco/models/reranker.py`
- [ ] Implement `RetrievalReranker` class
- [ ] Create `scripts/generate_reranker_data.py`
- [ ] Generate 50K-100K training pairs
- [ ] Create `scripts/train_reranker.py`
- [ ] Train model (contrastive ranking loss)
- [ ] Evaluate on held-out test set
- [ ] Integrate into RAG pipeline

## Acceptance Criteria

- [ ] Reranker trained and saved
- [ ] Improves retrieval accuracy over baseline
- [ ] Integrated into `klareco/rag/retriever.py`

## Documentation

See [[Deep-Learning-Models-Reference]] for model specs.

**Comment by marctjones** [comment]
Phase 6 cleanup - see CURRENT_PLAN.md for active work
